src/hotspot/share/runtime/mutex.hpp

```diff
@@ -81,8 +82,14 @@ class Mutex : public CHeapObj<mtSynchronizer> {
        native         = max_nonleaf    +   1
   };
 
+ private:
+  // The _owner field is only set by the current thread, either to itself after it has acquired
+  // the low-level _lock, or to NULL before it has released the _lock. Accesses by any thread other
+  // than the lock owner are inherently racy.
+  Thread* volatile _owner;
+  void raw_set_owner(Thread* new_owner) { Atomic::store(&_owner, new_owner); }
```

Comment:
set_owner_relaxed might be a better name.  Your call...

Reply:
Don't like 'relaxed'.  I'm not a huge fan of raw, since it's not raw here, but leave it.  It's fine.

Reply:
Thanks for restoring 'volatile.